optaplanner-core/src/main/java/org/optaplanner/core/impl/phase/AbstractPhase.java

```diff
@@ -107,14 +107,12 @@ public void setAssertShadowVariablesAreNotStaleAfterStep(boolean assertShadowVar
     public void solvingStarted(DefaultSolverScope<Solution_> solverScope) {
         // bestSolutionRecaller.solvingStarted(...) is called by DefaultSolver
         termination.solvingStarted(solverScope);
-        phaseLifecycleSupport.fireSolvingStarted(solverScope);
```

Comment:
Due to this, solvingStarted event was fired for each phase. I have moved this to DefaultSolver. As a result, solvingStarted event fires exactly once per Solver#solve(solution). Same for solvingEnded.